src/hotspot/share/services/nmtCommon.hpp

```diff
@@ -47,11 +47,21 @@ const int NMT_TrackingStackDepth = 4;
 // A few common utilities for native memory tracking
 class NMTUtil : AllStatic {
  public:
+  // Check if index is a valid MEMFLAGS enum value
+  static inline bool flag_index_is_valid(int index) {
+    return index >= 0 && index < mt_number_of_types;
+  }
+
+  // Check if flag value is a valid MEMFLAGS enum value
```

Comment:
Small nit, can you please clarify that this includes "mtNone"?

Reply:
Good point!